internal/persistence/ent/client/sip.go

```diff
@@ -76,14 +90,11 @@ func (c *client) CreateSIP(ctx context.Context, s *datatypes.SIP) error {
 		return rollback(tx, newDBError(err))
 	}
 
-	// Update SIP with DB data, to get generated values (e.g. ID).
+	// Update SIP with DB data, to get generated values (e.g. ID). Manually set
+	// the uploader data because the dbs result doesn't include the user edge.
+	uploader := s.Uploader
```

Comment:
`s.Uploader` is not getting updated from the database when you create a new user, so I think `uploader` will be missing the `CreatedAt` value, and could have the wrong `Email` and `Name` if the `s.Uploader` values are different than the database values.

Reply:
I think it's updated. If we get the user from `readOIDCUser` it's transformed [here](https://github.com/artefactual-sdps/enduro/blob/dev/issue-1291-create-user-tx/internal/persistence/ent/client/user.go#L90) and assigned to the `s.Uploader` [here](https://github.com/artefactual-sdps/enduro/blob/dev/issue-1291-create-user-tx/internal/persistence/ent/client/sip.go#L73), if we create a new user it's transformed [here](https://github.com/artefactual-sdps/enduro/blob/dev/issue-1291-create-user-tx/internal/persistence/ent/client/user.go#L46) where the passed `s.Uploader` is also being updated.

Reply:
Oh, you're right.  I forgot that `createUser()` mutates the passed user. :+1: